akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorFlightRecorder.scala

```diff
@@ -5,35 +5,23 @@
 package akka.actor.typed.internal
 
 import akka.actor.ActorPath
-import akka.actor.typed.{ ActorSystem, Extension, ExtensionId }
+import akka.actor.typed.internal.jfr.JFRActorFlightRecorder
 import akka.annotation.InternalApi
-import akka.util.FlightRecorderLoader
 
 /**
  * INTERNAL API
  */
 @InternalApi
-object ActorFlightRecorder extends ExtensionId[ActorFlightRecorder] {
+object ActorFlightRecorder {
 
-  override def createExtension(system: ActorSystem[_]): ActorFlightRecorder =
-    FlightRecorderLoader.load[ActorFlightRecorder](
-      system,
-      "akka.actor.typed.internal.jfr.JFRActorFlightRecorder",
-      NoOpActorFlightRecorder)
-}
+  lazy val recorder: ActorFlightRecorder = new JFRActorFlightRecorder()
```

Comment:
I think we can make the object the implementation instead of a separate singleton field, and then use that directly instead of a field in the classes that use it.